File: problems_list/0343.整数拆分/0343-整数拆分.js
/**
 * @param {number} n
 * @return {number}
 */
var integerBreak = function(n) {
    if(n==2) return 1
    if(n==3) return 2

    if(n%3==0) return Math.pow(3, Math.floor(n/3))
    var ans=1
    while(n>3){
        ans*=3
        n-=3
    }
    return n==1?Math.floor(ans/3)*4:ans*2
};

File: problems_list/0343.整数拆分/0343-整数拆分.cpp
class Solution {
public:
    int integerBreak(int n) {
        if (n <= 3) {
            return n - 1;
        }
        int quotient = n / 3;
        int remainder = n % 3;
        if (remainder == 0) {
            return (int)pow(3, quotient);
        } else if (remainder == 1) {
            return (int)pow(3, quotient - 1) * 4;
        } else {
            return (int)pow(3, quotient) * 2;
        }
    }
};